File: python/7 kyu/393.py
# https://www.codewars.com/kata/58fd52b59a9f65c398000096/train/python

# Passed

def min_and_max(l, d, x):
    get_sum_of_digits = lambda num: sum(int(digit) for digit in str(num))
    result = []
    for num in range(l, d + 1):
        sum_of_digits = get_sum_of_digits(num)
        if sum_of_digits == x:
            result.append(num)
            
    minimal_int, maximal_int = result[0], result[-1]
    result = [minimal_int, maximal_int]
    return result

output = min_and_max(100, 200, 10)
print(output)

File: nodejs/7 kyu/393.js
// https://www.codewars.com/kata/6912508732aab96c59c09c7d/train/javascript

// Passed

function reloadSheeps(arr) {
  let sheepSet = new Set("sheep");

  let result = [];
  for (const possibleSheep of arr) {
    let possibleSheepMap = new Object();
    for (const char of possibleSheep) {
      if (!sheepSet.has(char)) {
        continue;
      }

      if (!(char in possibleSheepMap)) {
        possibleSheepMap[char] = 0;
      }
      possibleSheepMap[char]++;
    }

    let isProperSheep = true;
    for (const char of sheepSet) {
      let charCountToCompare = char === "e" ? 2 : 1;
      if (possibleSheepMap[char] !== charCountToCompare) {
        isProperSheep = false;
        break;
      }
    }
    if (isProperSheep) {
      result.push("sheep");
    }
  }

  return result;
}

const output = reloadSheeps(["pe", "hehe", "heeps", "eee", "ti", "peehs"]);
console.log(output);